xdsl/irdl/irdl.py

```diff
@@ -572,7 +572,7 @@ class AttrSizedOperandSegments(AttrSizedSegments):
     See `AttrSizedSegments` for more information.
     """
 
-    attribute_name = "operand_segment_sizes"
+    attribute_name = "operandSegmentSizes"
```

Comment:
This is still my inital minimal change; do we want to follow the same convention change for others? as in `result_segment_sizes`

Context note: MLIR did this convention change meanwhile, but other segment sizes are an xDSL exclusivity, so no interoperability issues there.

Reply:
I like following a convention and I think we will usually be forced to adapt what MLIR is doing.

That being said, I am also against following a convention if it's annoying, so I am fine with leaving as-is to not introduce unnecessary work.

Reply:
Let's do it as followup then; it's rather trivial to do but I'd like to avoid bloating this PR's diff :slightly_smiling_face: 
And now that I think of it, because AFAIK it's unused out of tests, it could be done independently :thinking: 
I'll see if I can just pop a quick parallel PR for it.